### DISCUSSION TITLE
Passing Along Some Info

### DISCUSSION THREAD

**Discussion by hay-kot** [sysadminsmedia/homebox#35] [General]
Hey!

I wanted to pass along some thoughts I have/had on continuing development on Homebox. Looks like this is the most viable for currently, so I'll write about where I was at here. Hopefully, this comes across as me just sharing info and not trying to tell you folks what to do!

## Merging Locations and Items

- https://github.com/hay-kot/homebox/issues/696

I would highly encourage you guys to take a loop at this proposal, it solves a lot of features requests that already exists and reduces the maintenance of managing both Locations and Items separately. It also would allow users to customize display for different types of items, another big feature request.

It's a lot of upfront work but would likely reduce the future asks of "Can locations do x" or "Can we have _X_ type that's slightly different".

Implementing this would likely close https://github.com/sysadminsmedia/homebox/issues/27 as well.

## Password Reset

- https://github.com/hay-kot/homebox/pull/832

This PR has a almost finished implementation of Password Reset. I believe that last piece I needed was to actually implement the API call to do the reset once a token was issued. Probably worth taking a look at, if it's something you want to incorporate.

## Build Errors and Linter Errors

- https://github.com/hay-kot/homebox/pull/832

Also in the Passwords PR is a bunch of fixes for some changes that happened when I upgraded Golang CI lint that caused some breakages and errors to pop up. If you run into something similar it might be a good reference to fix some of those problems.

There's also some version bumps and CI fixes in there as well. 

--

Hope that's at least a little helpful!

**Comment by tankerkiller125** [comment]
@hay-kot Huge thanks for the information!

We'll for sure be looking at the things you've pointed out! I think we'll probably start with password resets :) The shared DB for items and locations for sure seems like it could be complex and a lot of work, but might be worth it.

**Comment by mcarbonne** [comment]
Totally agree with @hay-kot proposal to merge items and locations. In addition to enabling the creation of new concepts (containers...), it will also most likely solve the following issues : #110, #51, #38, #9.
